frontend/src/scenes/experiments/ExperimentView/components.tsx

```diff
@@ -191,22 +195,52 @@ export function PageHeaderCustom(): JSX.Element {
+                                        let target: HTMLElement | null = e.target as HTMLElement
+                                        while (target) {
+                                            if (target.id == 'start-date-picker-container') {
+                                                return
+                                            }
+                                            target = target.parentElement
+                                        }
+                                        setIsThreeDotMenuVisible(false)
+                                    }}
                                     overlay={
                                         <>
                                             <LemonButton
-                                                onClick={() => loadExperimentResults(true)}
+                                                onClick={() => {
+                                                    loadExperimentResults(true)
+                                                    setIsThreeDotMenuVisible(false)
+                                                }}
                                                 fullWidth
                                                 data-attr="refresh-experiment"
                                             >
                                                 Refresh experiment results
                                             </LemonButton>
                                             <LemonButton
-                                                onClick={() => loadSecondaryMetricResults(true)}
+                                                onClick={() => {
+                                                    loadSecondaryMetricResults(true)
+                                                    setIsThreeDotMenuVisible(false)
+                                                }}
                                                 fullWidth
                                                 data-attr="refresh-secondary-metrics"
                                             >
                                                 Refresh secondary metrics
                                             </LemonButton>
+                                            <StartDatePicker
+                                                currentStartDate={experiment.start_date!}
+                                                updateStartDate={(newStartDate: string) => {
+                                                    setIsThreeDotMenuVisible(false)
+                                                    updateExperiment({ start_date: newStartDate })
+                                                        .then(() => loadExperiment())
+                                                        .catch((error) =>
+                                                            console.error('error on loading experiment:', error)
+                                                        )
```

Comment:
We use kea.js for state management, which allows us to abstract the logic away from the UI components. This part mixes UI state management (setIsThreeDotMenuVisible(false)) and business logic (updating the experiment's start date and handling the success or failure of that update).

All you need to do here is call `updateExperiment`. Kea will reload the experiment after a successful update or handle any errors returned from the API call.